porch/pkg/git/git.go

```diff
@@ -520,7 +560,11 @@ func (r *gitRepository) discoverFinalizedPackages(ctx context.Context, ref *plum
 
 	var result []repository.PackageRevision
 	for _, krmPackage := range krmPackages.packages {
-		packageRevision, err := krmPackage.buildGitPackageRevision(ctx, revision, ref)
+		description, err := getPkgDescription(commit, krmPackage.path)
```

Comment:
This seems to only check the annotation on the latest commit message. Is that always correct? In particular for packages on the main branch, I think we need to be prepared for there being unrelated commits after the latest commit for the package. Can we use any of the `findLatestPackageCommit` or `packageRevisionHistoryIterator` functions for this?

Reply:
I updated `getPkgWorkspace` so that it uses `findLatestPackageCommit` when we are on the main branch.